### PR TITLE
[python] Read LeRobot video datasets for training

### PR DESCRIPTION
**Pull request by XiaoHongbo-Hope**
### Purpose

Read video-backed LeRobot v3 table groups through `PaimonLeRobotDataset`.

### Changes

- decode `.video` frame descriptors with TorchCodec or PyAV
- reuse bounded decoder caches per DataLoader worker
- build a lightweight packet timestamp/keyframe index for cold PyAV random reads, then seek directly to the target GOP
- support existing Episode, delta-window, padding, transform, and `return_uint8` behavior

TorchCodec is preferred; PyAV is the fallback.

### Tests

83 LeRobot/video tests passed, including real MP4, partial delta windows, B-frame random rereads, cold random-read decode bounds, optional-dependency gates, and spawn workers.

### CODE REVIEW

File: paimon-python/pypaimon/multimodal/lerobot/dataset.py [hunk 1]
```diff
@@ -1164,6 +1224,173 @@ def _image_tensor(payload, feature, return_uint8=False):
...
+            raise IndexError("Video frame index %d is out of range." % index)
+        frame = self._cache.pop(index, None)
+        if frame is not None:
+            self._cache[index] = frame
+            return self._tensor(frame)
+
+        at_frontier = self._next_index == len(self._timestamps)
+        if index != self._next_index and not (
+                at_frontier and index >= self._next_index):
+            self._seek(index)
```
Comment (JingsongLi) [inline paimon-python/pypaimon/multimodal/lerobot/dataset.py:1315]: [P2] Avoid decoding the full video prefix on cold random reads

When PyAV is selected explicitly or used as the fallback, a fresh decoder skips seeking and decodes every frame from 0 through the first requested index. The collator retains only eight videos per camera, so shuffled training over more files can repeatedly discard the learned timestamp/keyframe index and pay this cost again.

On a 6,000-frame, 62 MB H.264 MP4, reading frame 5,999 decoded all 6,000 frames and took about 1.4 seconds. Scanning packet timestamps took 12–15 ms, and seeking to the final GOP and decoding its 30 frames took about 8 ms.

Please build or reuse an ordinal/keyframe index from demuxed packets so cold random reads can seek to the required GOP while preserving exact frame ordinals. The existing importer already uses packet timestamps to determine frame ordinals.

Comment (XiaoHongbo-Hope) [reply paimon-python/pypaimon/multimodal/lerobot/dataset.py]: Fixed in 4cfabc7b34. A cold non-zero read now demuxes packet timestamps/keyframe flags once, builds the complete ordinal/keyframe index, seeks to the target GOP, and decodes only from that keyframe. The new regression requests frame 95 from a fresh 120-frame decoder and verifies only frames 90–95 are decoded; I also reran random reads on a real MP4 with B-frames.

**Comment by JingsongLi** [comment]
+1
